dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/CDIContainerImpl.java

```diff
@@ -224,7 +221,15 @@ public void applicationStarted(Application application) throws CDIException {
                 if (modules != null) {
                     for (BeanDeploymentModule module : modules) {
                         if (!module.isWebModule()) {
-                            eventManager.fireStartupEvent(module);
+                            String id = module.getId();
+                            WebSphereBeanDeploymentArchive bda = deployment.getBeanDeploymentArchive(id);
```

Comment:
Do we have any guarantee that the BDA for the module has the same ID as the `BeanDeploymentModule`?

Reply:
The ID in the BeanDeploymentModule and the BDA are both built from the same String

BDAFactory.java line 84
`EEModuleDescriptor eeModuleDescriptor = new WebSphereEEModuleDescriptor(archiveID, extensionArchive.getType());`

This sets the EEModuleDescriptor's id to archiveID And eventually the BeanDeploymentModule gets its ID from the EEModuleDescriptor's id.

`module = new BeanDeploymentModule(descriptor.getId()` in weld's class BeanDeploymentModules

Then a few lines later in BDAFactory.java we pass `archiveID` into the constructor BeanDeploymentModule where it becomes the ID. 

The exception is for library modules in `CDIContainerImpl.processModuleLibraries` where the module descriptor will have the ID of the parent. So for those we'll be setting our TCCL and CMD based on the library's parent.